string part_type
---
bool success
